openwisp_monitoring/monitoring/api/views.py

```diff
@@ -0,0 +1,121 @@
+import logging
+from collections import OrderedDict
+
+from django.conf import settings
+from django.http import HttpResponse
+from pytz import timezone as tz
+from pytz.exceptions import UnknownTimeZoneError
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from swapper import load_model
+
+from ...monitoring.exceptions import InvalidChartConfigException
+
+logger = logging.getLogger(__name__)
+Chart = load_model('monitoring', 'Chart')
+Metric = load_model('monitoring', 'Metric')
+Organization = load_model('openwisp_users', 'Organization')
+
+
+class DashboardTimeseriesView(APIView):
+    def _get_organization_lookup(self):
+        query_orgs = self.request.query_params.get('organization_slug', '')
+        if query_orgs:
+            query_orgs = query_orgs.split(',')
+            org_ids = Organization.objects.filter(slug__in=query_orgs).values_list(
+                'id', flat=True
+            )
+            if len(org_ids) != len(query_orgs):
+                raise NotFound
+            if self.request.user.is_superuser or all(
+                [str(id) in self.request.user.organizations_managed for id in org_ids]
+            ):
+                orgs = org_ids
+            else:
+                raise PermissionDenied
+        else:
+            if self.request.user.is_superuser:
+                orgs = []
+            elif self.request.user.organizations_managed:
+                orgs = self.request.user.organizations_managed
+            else:
+                raise PermissionDenied
+
+        if not orgs:
+            return ''
+        org_lookup = []
+        for org in orgs:
+            org_lookup.append(f"organization_id = '{org}'")
+        return 'AND ({org_lookup})'.format(org_lookup=' OR '.join(org_lookup))
+
+    def get(self, request):
+        org_lookup = self._get_organization_lookup()
```

Comment:
This method is similar to the method in device_monitoring API views. We can make something reusable.